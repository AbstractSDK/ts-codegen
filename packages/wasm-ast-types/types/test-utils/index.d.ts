export declare const expectCode: (ast: any) => void;
export declare const printCode: (ast: any) => void;
