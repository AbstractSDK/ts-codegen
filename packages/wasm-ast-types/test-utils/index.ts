import generate from '@babel/generator';
import { sync as glob } from 'glob';
import { readFileSync } from 'fs';
import { join } from 'path';
import { JSONSchema } from '../src/types';

export const expectCode = (ast) => {
    expect(
        generate(ast).code
    ).toMatchSnapshot();
}

export const printCode = (ast) => {
    console.log(
        generate(ast).code
    );
}

interface GlobContract {
    name: `/${string}.json`;
    content: JSONSchema;
}

export const globContracts = (p: string): GlobContract[] => {
    return glob(join(__dirname, '/../../../__fixtures__/', p, '/*.json')).map(file => {
        return { name: file.split('__fixtures__')[1], content: JSON.parse(readFileSync(file, 'utf-8')) }
    });
};
