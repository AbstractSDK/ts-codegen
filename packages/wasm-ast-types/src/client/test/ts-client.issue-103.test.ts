import contract from '../../../../../__fixtures__/issues/103/schema.json';

import {
    createQueryClass,
    createExecuteClass,
    createExecuteInterface,
    createTypeInterface
} from '../client'
import { expectCode, printCode, makeContext } from '../../../test-utils';

const message = contract.query
const ctx = makeContext(message);

it('execute_msg_for__empty', () => {
    expectCode(createTypeInterface(
        ctx,
        message
    ))
})


it('query classes', () => {
    expectCode(createQueryClass(
        ctx,
        'QueryClient',
        'ReadOnlyInstance',
        message
    ))
});

// it('query classes response', () => {
//     expectCode(createTypeInterface(
//         ctx,
//         contract.responses.all_debt_shares
//     ))
// });

it('execute classes array types', () => {
    expectCode(createExecuteClass(
        ctx,
        'Client',
        'Instance',
        null,
        message
    ))
});

it('execute interfaces no extends', () => {
    expectCode(createExecuteInterface(
        ctx,
        'SG721Instance',
        null,
        message
    ))
});
