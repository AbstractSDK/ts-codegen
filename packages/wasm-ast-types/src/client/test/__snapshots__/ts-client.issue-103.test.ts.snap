// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute classes array types 1`] = `
"export class Client implements Instance {
  client: SigningCosmWasmClient;
  sender: string;
  contractAddress: string;

  constructor(client: SigningCosmWasmClient, sender: string, contractAddress: string) {
    this.client = client;
    this.sender = sender;
    this.contractAddress = contractAddress;
  }

}"
`;

exports[`execute interfaces no extends 1`] = `
"export interface SG721Instance {
  contractAddress: string;
  sender: string;
}"
`;

exports[`execute_msg_for__empty 1`] = `"export type QueryMsg = QueryMsg;"`;

exports[`query classes 1`] = `
"export class QueryClient implements ReadOnlyInstance {
  client: CosmWasmClient;
  contractAddress: string;

  constructor(client: CosmWasmClient, contractAddress: string) {
    this.client = client;
    this.contractAddress = contractAddress;
  }

}"
`;
